tables/leaf.py

```diff
@@ -782,6 +853,117 @@ def flush(self) -> None:
 
         self._g_flush()
 
+    def chunk_info(self, coords: tuple[int, ...]) -> ChunkInfo:
+        """Get storage information about the chunk containing the `coords`.
+
+        The coordinates `coords` are a tuple of integers with the same rank as
+        the dataset.
+
+        Return a :class:`ChunkInfo` instance with the information.
+
+        The coordinates need not be aligined with chunk boundaries.  This
+        means that this method may be used to get the start coordinates of the
+        chunk that contains the item at the given coordinates, for use with
+        other direct chunking operations (see :attr:`ChunkInfo.start`).
+
+        If the coordinates are within the dataset's shape but there is no such
+        chunk in storage (missing chunk), a :class:`ChunkInfo` with a valid
+        ``start`` and ``filter_mask = offset = size = None`` is returned.  If
+        the coordinates are beyond the shape, :exc:`IndexError` is raised
+        (even if the start of the chunk would fall within the shape).
+
+        Calling this method on a non-chunked dataset raises a
+        :exc:`NotChunkedError`.
+
+        """
+        self._check_chunked()
+        self._check_chunk_within(coords)
+
+        coords = np.array(coords, dtype=SizeType)
+        filter_mask, offset, size = self._g_chunk_info(coords)
+
+        # Align coordinates to chunk boundary.
+        chunkshape = self.chunkshape
+        coords //= chunkshape
+        coords *= chunkshape
+        return ChunkInfo(tuple(coords), filter_mask, offset, size)
+
+    def read_chunk(self, coords: tuple[int, ...],
+                   out: Optional[Union[bytearray, NPByteArray]]=None,
```

Comment:
In Python v3.10 you can use `Union[bytearray, NPByteArray] | None` if you prefer

Reply:
Ditto in the comment above.